service/dash.go

```diff
@@ -32,7 +39,7 @@ func GenerateDash(fileName string) {
 
 	generateAudioFiles(targetFile, outputPath, &wg)
 
-	generateVideoFiles(targetFile, outputPath, &wg)
+	generateVideoFiles(targetFile, outputPath, watermarkFile, withWatermark, watermark, &wg)
```

Comment:
could you pass the struct in this function and restructure only where you are using it

Reply:
done